docs/reference/patterns.md

```diff
@@ -0,0 +1,58 @@
+## Pattern detection and metadata enhancement
+
+The PatternDetector is responsible for identifying specific patterns within the input, such as using regular expressions to detect ISSN, ISBN, DOI, and PMID. Other techniques than regular expressions may also occur here, such as doing phrase matching to identify known scientific journals, or fingerprint matching to identify librarian curated responses.
```

Comment:
I think `Pattern Detector` is a more descriptive name for what the model and GraphQL type are trying to do, but less descriptive for the GraphQL field than `StandardIdentifiers`. Would it be confusing for the field to have a different name than the model and type? Is that even possible?

This documentation also mentions an `Enhancer`, which is not yet implemented in this PR. (I assume it will be soon, though, based on the experiment you shared last week.) If that model/type is not named `Enhancer`, then an alternative approach could be to clarify which models/types are an example of `Pattern Detector`, and which are an example of `Enhancer`. In other words, `Pattern Detector` and `Enhancer` are how we're describing the features at a high level, whereas `StandardIdentifiers` and `RecordsFound` (or whatever) are the actual implementations of those features. In that case, we should make that clarification in this reference doc.

I do think we should make a decision on this before merging, because I suspect it will be confusing if the naming conventions in the docs don't match what's in the code.

Reply:
This is super helpful feedback and I agree. I'll give it some thought, likely ask for a synchronous chat, then make appropriate changes.

Reply:
I'm leaning towards having PatternDetectors and Enhancers be the high level concept for this type of work, and StandardIdentifiers being our first implementation of a PatternDetector and coming soon, LookupDoi (etc) being implementations of Enhancers.

I'll be adding a commit that reflects this and then reaching out for further discussion.